pkg/skaffold/build/tag/tag.go

```diff
@@ -29,18 +30,22 @@ type ImageTags map[string]string
 // Tagger is an interface for tag strategies to be implemented against
 type Tagger interface {
 	// GenerateTag generates a tag for an artifact.
-	GenerateTag(workingDir, imageName string) (string, error)
+	GenerateTag(workingDir string, image *latest.Artifact) (string, error)
 }
 
 const DeprecatedImageName = "_DEPRECATED_IMAGE_NAME_"
 
 // GenerateFullyQualifiedImageName resolves the fully qualified image name for an artifact.
 // The workingDir is the root directory of the artifact with respect to the Skaffold root,
 // and imageName is the base name of the image.
-func GenerateFullyQualifiedImageName(t Tagger, workingDir, imageName string) (string, error) {
+func GenerateFullyQualifiedImageName(t Tagger, workingDir string, image *latest.Artifact) (string, error) {
```

Comment:
done, though this one was a little strange because `GenerateTag` is entangled with a few functions that already use `*latest.Artifact`. did the best I could and I think it's a little cleaner now